### DISCUSSION TITLE
Combining Audio Streams

### DISCUSSION THREAD

**Discussion by ClockworkPirate** [naudio/NAudio#912] [General]
I'm not sure if this is the right place to ask but I've been struggling with an audio problem fort quite a while, and I suspect NAudio might be my saviour. Basically I'm wrtiting a VR karaoke application and I need to combine the audio from the Mic and also from a audio live stream into a single audio data stream which I can then pipe across the networking platform I'm using.

The application is written in Unity (so .Net/Mono) and I believe this is something I can use (https://github.com/yhchen/NAudio-Unity)

The data I'm seeing in my VR application appears to be float arrays, but I haven't a clue where to start. I was wondering if there's any way to get any guidance?

Thanks a million in advance.
Martin

**Comment by Axwabo** [comment]
Have a look at [OnAudioFilterRead hook](https://docs.unity3d.com/6000.3/Documentation/ScriptReference/MonoBehaviour.OnAudioFilterRead.html) in which you could read from a `MixingSampleProvider`
Not sure how capturing the mic would work tbh
